### PR TITLE
Creation of `qibo.models.encodings` and implementation of `unary_encoder`

### PR DESCRIPTION
**Pull request by renatomello**
Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Review by MatteoRobbiati — APPROVED** [review]
Thank you, it is fine for me. 
Just a question, I noticed the notation is slightly different from the paper you cite in the docstrings. Did you apply these modifications in order to correctly deploy the data loader?
It perfectly works btw, so it sounds good to me.

**Comment by renatomello** [comment]
> Thank you, it is fine for me. Just a question, I noticed the notation is slightly different from the paper you cite in the docstrings. Did you apply these modifications in order to correctly deploy the data loader? It perfectly works btw, so it sounds good to me.

I don't know which part exactly you're referring to, but it's probably just personal preference.

**Review by stavros11 — APPROVED** [review]
Minor comments, looks good to merge as it is. The only thing that is slightly weird is that this is exposed as a model, but the name follows different convention than other models. Some users may expect `from qibo.models import UnaryEncoder`. I get that you are following the function naming convention so it should be fine to keep as it is.

File: src/qibo/models/encodings.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    nqubits = len(data)
+    j_max = int(nqubits / 2)
+
+    circuit, _ = _generate_rbs_pairs(nqubits)
+
+    # calculating phases and setting circuit parameters
+    r_array = np.zeros(nqubits - 1, dtype=float)
+    phases = np.zeros(nqubits - 1, dtype=float)
+    for j in range(1, j_max + 1):
```
[the remark this answers is not in the document]
Comment (renatomello) [reply src/qibo/models/encodings.py:62]: This algorithm is written intentionally done like this in the paper in order to be $\mathcal{O}(d)$

File: src/qibo/models/encodings.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    for j in range(j_max - 1, 0, -1):
+        r_array[j - 1] = math.sqrt(r_array[2 * j] ** 2 + r_array[2 * j - 1] ** 2)
+        phases[j - 1] = math.acos(r_array[2 * j - 1] / r_array[j - 1])
+
+    circuit.set_parameters(phases)
+
+    return circuit
+
+
+def _generate_rbs_pairs(nqubits):
```
[the remark this answers is not in the document]
Comment (renatomello) [reply src/qibo/models/encodings.py:76]: Thank you

**Pull request merged**